src/cascade/dismod/serialize.py

```diff
@@ -138,17 +139,21 @@ def make_log_table():
     )
 
 
-def rec_build_nodes_table(node, parent):
-    parent_id = parent.id if parent is not None else np.NaN
-    result = [{"node_name": node.info["location_name_short"], "c_location_id": node.id, "parent": parent_id}]
-    for child in node.level_n_descendants(1):
-        result += rec_build_nodes_table(child, node)
-    return result
+def _parent_of(location_graph, node):
+    try:
+        return next(location_graph.predecessors(node))
+    except StopIteration:
+        return np.nan
```

Comment:
This is used just below. I could make it shorter with a nested comprehension, but this was so much clearer.